ur_kinematics/src/ur_moveit_plugin.cpp

```diff
@@ -325,22 +325,9 @@ bool URKinematicsPlugin::initialize(const std::string &robot_description,
 
   private_handle.param<std::string>("arm_prefix", arm_prefix_, "");
 
-  ur_joint_names_.push_back(arm_prefix_ + "shoulder_pan_joint");
-  ur_joint_names_.push_back(arm_prefix_ + "shoulder_lift_joint");
-  ur_joint_names_.push_back(arm_prefix_ + "elbow_joint");
-  ur_joint_names_.push_back(arm_prefix_ + "wrist_1_joint");
-  ur_joint_names_.push_back(arm_prefix_ + "wrist_2_joint");
-  ur_joint_names_.push_back(arm_prefix_ + "wrist_3_joint");
-
-  ur_link_names_.push_back(arm_prefix_ + "base_link");       // 0
-  ur_link_names_.push_back(arm_prefix_ + "ur_base_link");    // 1
-  ur_link_names_.push_back(arm_prefix_ + "shoulder_link");   // 2
-  ur_link_names_.push_back(arm_prefix_ + "upper_arm_link");  // 3
-  ur_link_names_.push_back(arm_prefix_ + "forearm_link");    // 4
-  ur_link_names_.push_back(arm_prefix_ + "wrist_1_link");    // 5
-  ur_link_names_.push_back(arm_prefix_ + "wrist_2_link");    // 6
-  ur_link_names_.push_back(arm_prefix_ + "wrist_3_link");    // 7
-  ur_link_names_.push_back(arm_prefix_ + "ee_link");         // 8
+  // Load the joint names from the group model
+  ur_joint_names_ = joint_model_group->getJointModelNames();
+  ur_link_names_ = joint_model_group->getLinkModelNames();
 
   ur_joint_inds_start_ = getJointIndex(ur_joint_names_[0]);
```

Comment:
@gavanderhoorn This variable is used to identify the start position in a given joint vector for the purposes of transforming input poses to the solver's base frame. With these changes as they are right now, you can see that this will always be set to zero (assuming that the solver starts at the group's base frame). Previously (currently), it could be non-zero.

I hope I'm right at least =P